### DISCUSSION TITLE
Unable to get blocking rule right

### DISCUSSION THREAD

**Discussion by agneevX** [AdguardTeam/AdGuardHome#3235] [Q&A]
I'm trying to block `graph.facebook.com` on all clients except a specific few using this rule:

```regex
||graph.facebook.com^$important,$client=~client-1|~client-2|~client-3
```

However that does not seem to work. 

I've also tried

```regex
||graph.facebook.com^
@@||graph.facebook.com^$important,$client=client-1|client-2|client-3
```

**Comment by ameshkov** [comment]
The second `$` is unnecessary.

`$` is a delimiter between the rule pattern (the one that matches the domain name) AND the comma-separated list of options.

Should be something like this:
`||graph.facebook.com^$important,client=~client-1|~client-2|~client-3`

**Comment by agneevX** [comment]
> Should be something like this:
||graph.facebook.com^$important,client=~client-1|~client-2|~client-3

@ameshkov This does not seem to work. Clients that are requesting this URL and mentioned here seem to get blocked.

The only way this seems to work is if I add a block rule and then an unblock rule:

```regex
||*graph.facebook.com^
@@||*graph.facebook.com^$important,client=client-1|client-2
```

EDIT: Now it seems that this is not working either (whietlisted clients' queries getting blocked)

**Comment by ainar-g** [comment]
```
@@||graph.facebook.com^$client=client-name-1|client-name-2
```

This works for me.  Are you sure that you're entering the **names** (as opposed to ClientIDs) of the clients correctly?

**Comment by agneevX** [comment]
Sorry for the delay. This seems to not be working for me.

> Are you sure that you're entering the names (as opposed to ClientIDs) of the clients correctly?

Yes I am.

Since `graph.facebook.com` is not in any blocklist, do I need to add a single rule that blocks the website for every device but the ones I specify or do I need to add two rules: One would block it for all, the other would whitelist it for x clients?

**Reply by ameshkov** [reply]
Yes, a single rule is better
